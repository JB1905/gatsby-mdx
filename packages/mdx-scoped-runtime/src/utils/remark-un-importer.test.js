import mdx from '@mdx-js/mdx';
import plugin from './remark-un-importer';

it('should not change simple mdx', async () => {
  const MDX = `
# Hello world

- item 1
- item 2`;

  const result = await mdx(MDX, { mdPlugins: [plugin] });
  expect(result).toEqual(await mdx(MDX));
});

it('should remove imports', async () => {
  const MDX = `import { Divider } from "../../components";

# Hello world

<Divider />

- item 1
- item 2`;

  const result = await mdx(MDX, { mdPlugins: [plugin] });
  expect(result).not.toEqual(await mdx(MDX));
  expect(result).toMatchSnapshot();
});
