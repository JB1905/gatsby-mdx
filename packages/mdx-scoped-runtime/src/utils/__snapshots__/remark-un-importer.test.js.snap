// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should remove imports 1`] = `
"

export default ({components, ...props}) => <MDXTag name=\\"wrapper\\"  components={components}><MDXTag name=\\"h1\\" components={components}>{\`Hello world\`}</MDXTag>
<Divider />
<MDXTag name=\\"ul\\" components={components}>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 1\`}</MDXTag>
<MDXTag name=\\"li\\" components={components} parentName=\\"ul\\">{\`item 2\`}</MDXTag>
</MDXTag></MDXTag>"
`;
